### PR TITLE
Replace lock_table config option with dynamodb_table

### LINKED ISSUE #269: lock_table deprecated
The terraform remote state configuration directive `lock_table` for the S3 backend  is deprecated in favour of the new directive `dynamodb_table`

terraform throws the following warning when initialising the backend:

```
Initializing the backend...
"lock_table": [DEPRECATED] please use the dynamodb_table attribute
```

I tried using `dynamodb_table` in the terragrunt remote config but it is not recognised.

### PR DESCRIPTION
**Pull request by robinbowes**
Fixes #269.

Not tested at all.

Please review and offer feedback.

### CODE REVIEW

**Comment by mcalhoun** [comment]
Thanks for the PR @robinbowes! It looks good to me, but you'll probably also want to update the [test](https://github.com/gruntwork-io/terragrunt/blob/1b0cb787c0314681e92f5ff083486eb5a988452f/test/integration_test.go#L523) as well

**Comment by robinbowes** [comment]
Good spot. How's that look? If it wasn't 02:30 here I'd think about setting up a local go env and running the tests :)

File: remote/remote_state_s3.go [hunk 1]
```diff
@@ -15,12 +15,12 @@ import (
...
-	Key       string `mapstructure:"key"`
-	Region    string `mapstructure:"region"`
-	Profile   string `mapstructure:"profile"`
-	LockTable string `mapstructure:"lock_table"`
+	Encrypt       bool   `mapstructure:"encrypt"`
+	Bucket        string `mapstructure:"bucket"`
+	Key           string `mapstructure:"key"`
+	Region        string `mapstructure:"region"`
+	Profile       string `mapstructure:"profile"`
+	DynamoDBTable string `mapstructure:"dynamodb_table"`
```
Comment (brikis98) [inline remote/remote_state_s3.go]: I'm a little concerned about dropping the old param. Ideally, we'd follow Terraform's lead and report a deprecation warning for at least a little while. Or, even simpler, Terragrunt should accept both params and pass them on to Terraform unchanged, and Terraform itself would log the deprecation warning.

Comment (robinbowes) [reply remote/remote_state_s3.go]: Yeah, I agree, but that's more work :)

I'll see if I can make some time to figure that out soon.

R.

Comment (chapati23) [reply remote/remote_state_s3.go]: not a go expert but can't you just keep both params?

```go
LockTable string `mapstructure:"lock_table"`
DynamoDBTable string `mapstructure:"dynamodb_table`"
```

Comment (brikis98) [reply remote/remote_state_s3.go]: Yup, keeping both is most likely the way to go. Also need to check both in the `createLockTableIfNecessary` method.

**Comment by brikis98** [comment]
@robinbowes Any update on this?

**Comment by robinbowes** [comment]
I'm afraid I've not had time to re-visit yet.

**Comment by brikis98** [comment]
OK, I just opened #309 with the full fix, which also allows me to run tests in the PR. Closing this one now. Thanks!

**Pull request closed without merging**